jobs/periodic/delete-unreachable-achievements/courses.ts

```diff
@@ -0,0 +1,78 @@
+import { getLogger } from '../../../common/logger/logger';
+import { EventRelationType, createRelation, parseRelation } from '../../../common/achievement/relation';
+import { prisma } from '../../../common/prisma';
+import tracer from '../../../common/logger/tracing';
+
+const logger = getLogger('CourseAchievementCleanup');
+
+const ParticipationGroupTemplate = 'pupil_course_participation';
+
+export const deleteUnreachableCourseAchievements = tracer.wrap('delete', _deleteUnreachableCourseAchievements);
+async function _deleteUnreachableCourseAchievements(dryRun: boolean = true) {
+    const till = new Date();
+    till.setDate(till.getDate() - 1);
+
+    const unachievedAchievementes = await prisma.user_achievement.findMany({
```

Comment:
Technically the relations would be enough, right? 

```
await prisma.user_achievement.findMany({
  distinct: ['relation'],
  select: {
    relation: true,
  },
})
```

That would reduce complexity by factor N :)

Reply:
We need all the user ids later in the process. I could use your suggestion, but that would mean making N additional queries, one for each course, to retrieve them.